src/ol/source/Raster.js

```diff
@@ -230,6 +230,19 @@ class RasterSource extends ImageSource {
       wantedTiles: {}
     };
 
+    this.setAttributions(function(frameState) {
+      let attributions = [];
+      options.sources.forEach(sourceOrLayer => {
+        const source = sourceOrLayer instanceof Source ? sourceOrLayer : sourceOrLayer.getSource();
+        const attributionGetter = source.getAttributions();
+        if (typeof attributionGetter === 'function') {
+          const sourceAttribution = attributionGetter(frameState);
+          attributions = [...attributions, ...sourceAttribution];
```

Comment:
Another ES6 feature we've avoided so far. Also, in terms of avoiding garbage, something like this might be more efficient:
```js
attributions.push.apply(attributions, sourceAttribution);
```